packages/neon/neon_files/lib/widgets/actions.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:filesize/filesize.dart';
+import 'package:flutter/material.dart';
+import 'package:neon/utils.dart';
+import 'package:neon_files/l10n/localizations.dart';
+import 'package:neon_files/neon_files.dart';
+import 'package:provider/provider.dart';
+
+class FileActions extends StatelessWidget {
+  const FileActions({
+    required this.details,
+    super.key,
+  });
+
+  final FileDetails details;
+
+  Future<void> onSelected(final BuildContext context, final FilesFileAction action) async {
+    final bloc = Provider.of<FilesBloc>(context, listen: false);
+    final browserBloc = bloc.browser;
+    switch (action) {
+      case FilesFileAction.toggleFavorite:
+        if (details.isFavorite ?? false) {
+          bloc.removeFavorite(details.path);
+        } else {
+          bloc.addFavorite(details.path);
+        }
+        break;
+      case FilesFileAction.details:
+        await Navigator.of(context).push(
+          MaterialPageRoute(
+            builder: (final context) => FilesDetailsPage(
+              bloc: bloc,
+              details: details,
+            ),
+          ),
+        );
+        break;
+      case FilesFileAction.rename:
+        final result = await showRenameDialog(
+          context: context,
+          title:
+              details.isDirectory ? AppLocalizations.of(context).folderRename : AppLocalizations.of(context).fileRename,
+          value: details.name,
+        );
+        if (result != null) {
+          bloc.rename(details.path, result);
+        }
+        break;
+      case FilesFileAction.move:
+        final b = bloc.getNewFilesBrowserBloc();
+        final originalPath = details.path.sublist(0, details.path.length - 1);
+        b.setPath(originalPath);
```

Comment:
I know you didn't introduce this code, but could you make it so that it is possible to pass an optional initial path to the files browser bloc instead of having to call setPath directly after it?